src/mito2/src/memtable.rs

```diff
@@ -110,6 +110,13 @@ pub trait Memtable: Send + Sync + fmt::Debug {
         predicate: Option<Predicate>,
     ) -> Result<BoxedBatchIterator>;
 
+    /// Returns the ranges in the memtable.
+    fn ranges(
+        &self,
+        projection: Option<&[ColumnId]>,
+        predicate: Option<Predicate>,
+    ) -> Vec<MemtableRange>;
```

Comment:
Maybe return a result
```suggestion
    ) -> Result<Vec<MemtableRange>>;
```

Reply:
Should we postpone this until some memtable implementations need to return results?